api_tests/lib/actor.ts

```diff
@@ -106,17 +106,44 @@ export class Actor {
         if (network != "regtest") {
             throw Error("Expected network regtest, found " + network);
         }
-        let type = action.type;
-
-        switch (type) {
+        switch (action.type) {
             case "bitcoin-send-amount-to-address": {
                 action.payload.should.include.all.keys("to", "amount");
                 let { to, amount } = action.payload;
 
                 return this.wallet.btc().sendToAddress(to, parseInt(amount));
             }
             case "bitcoin-broadcast-signed-transaction": {
-                let { hex } = action.payload;
+                action.payload.should.include.all.keys("hex", "locktime");
+
+                let fetchMedianTime = async () => {
+                    let blockchainInfo = await bitcoin.getBlockchainInfo();
+                    return blockchainInfo.mediantime;
+                };
+
+                let { hex, locktime } = action.payload;
+
+                // If it is equal, it is rejected by bitcoin-core :(
+                locktime += 1;
+
+                let medianTime = await fetchMedianTime();
+                let diff = locktime - medianTime;
+
+                if (locktime && diff > 0) {
+                    console.log(`Waiting for median time to pass %d`, locktime);
+
+                    while (diff > 0) {
+                        await sleep(1000);
+
+                        medianTime = await fetchMedianTime();
+                        diff = locktime - medianTime;
+
+                        console.log(
```

Comment:
❌  to be removed

Reply:
Why? Without these logs, your test suite hangs for an unknown amount of time and you don't know what is going on.

Reply:
What is the difference other than taste?

Reply:
> What is the difference other than taste?
Better coding style? I think the log vs. loop behavior makes more sense that way.

Currently:
* `sleep`
* `fetchMedianTime();`
* log `Current median time:          XX+~sleep time` 
* continue with/after the loop

While I would propose to: 

* log `Current median time:          XX`  --> you still have this value from before
* `sleep`
* `fetchMedianTime();`
* continue with/after the loop

i.e..current behavior will be

* _current_median_time=8_
* `Waiting for median time to pass 10.`
* *sleeping randomly unknown time...*
* `Current median time:                   9`
* *sleeping randomly unknown time...*
* .... continue with after loop

vs. 

* _current_median_time=8_
* `Waiting for median time to pass 10.`
* `Current median time:                   8`
* *sleeping randomly unknown time...*
* `Current median time:                   9`
* *sleeping randomly unknown time...*
* .... continue with after loop